### ISSUE TITLE
BIASCOR task failing during final cleanup stage

### ISSUE THREAD

**Issue by geotaylo** [dessn/Pippin#145]
I've only encountered this when trying to float the location of the mass step using the u7: 1 opt, but I'm not certain that it's correlated.

See log in $PIPPIN_OUTPUT/GT_MASSSPLIT_R1/6_BIASCOR/3YR_MassStep_fitsplit/output/SCRIPTS_BBCFIT/CPU0000_JOBLIST_midway2-login2.LOG

**Comment by OmegaLambda1998** [comment]
@RickKessler This looks to be an SNANA issue.

```
1  Write FAIL to done stamp file:
  2      ALL.DONE
  3 # merge_driver:      Remove BUSY_MERGE_CPU0000.LOCK for T_midnight=82428
  4 ['Check CPU0000*.LOG for merge crash']
  5 Traceback (most recent call last):
  6   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_bat    ch_jobs.py", line 329, in run_merge_driver
  7     program.merge_driver()
  8   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_base.py", line 1550, in merge_driver
  9     if cleanup_flag:  self.merge_cleanup_final()
 10   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_bbc.py", line 2067, in merge_cleanup_final
 11     self.make_fitpar_summary()
 12   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_bbc.py", line 2532, in make_fitpar_summary
 13     val = str(result[key].split()[0])
 14 TypeError: string indices must be integers
 15 string indices must be integers
 16 Traceback (most recent call last):
 17   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_bat    ch_jobs.py", line 329, in run_merge_driver
 18     program.merge_driver()
 19   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_base.py", line 1550, in merge_driver
 20     if cleanup_flag:  self.merge_cleanup_final()
 21   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_bbc.py", line 2067, in merge_cleanup_final
 22     self.make_fitpar_summary()
 23   File "/project2/rkessler/PRODUCTS/SNANA/SNANA/util/submit_batch/submit_pro    g_bbc.py", line 2532, in make_fitpar_summary
 24     val = str(result[key].split()[0])
 25 TypeError: string indices must be integers
 ```

As far as I can tell the issue comes with reading:
```
1 BBCFIT_RESULTS:
  2   - NSNFIT:         322     18
  3   - SIGINT:       0.10766    0.0
  4   - alpha0:       0.15482  0.00965
  5   - beta0:        2.77717  0.10463
  6   - gamma0:       0.09656  0.02137
  7   - HOST_LOGMASS  10.36286  0.11858
  ```
  
 Which is read using:
 ```
1             for result in BBCFIT_RESULTS:
   2                 #print(f" xxx result = {result}  key = {result.keys()} ")
   3                 for key in result:
   4                     val = str(result[key].split()[0])
   5                     err = str(result[key].split()[1])
   6                     KEY = f"{key}:"
   7                     f.write(f"    {KEY:<12}  {val:>8} +_ {err:<8} \n")
   ```
   
   But it's not clear to be how this results in a string index error?

**Comment by OmegaLambda1998** [comment]
The only thing I can think of is, for some reason `result` is a string rather than a dictionary. `BBCFIT_RESULTS` is created via:
```
1             bbc_yaml   = util.extract_yaml(YAML_FILE, None, None )
   2             BBCFIT_RESULTS = bbc_yaml['BBCFIT_RESULTS']
   ```
   
   So I would assume this is actually a bug in `util.extract_yaml`

**Comment by RickKessler** [comment]
in the output YAML written by SALT2mu.c, the key format was -12.12s  instead of just -12s ... the former truncates to 12 characters (HOST_LOGMASS) and thus dropped the required colon. The latter writes all characters even if they are longer than 12 characters:
"HOST_LOGMASS_cen:"

The fix is to replace -12.12s with -20s format in write_yaml_info().

**Comment by geotaylo** [comment]
Seems to be fixed now, thanks all.
